wallet/storage/src/internal/store_tx.rs

```diff
@@ -158,13 +161,16 @@ macro_rules! impl_read_ops {
                     .map(Iterator::collect)
             }
 
-            fn exactly_one_root_key(&self) -> crate::Result<bool> {
+            fn check_root_keys_sanity(&self) -> crate::Result<()> {
                 self.storage
                     .get::<db::DBRootKeys, _>()
                     .prefix_iter_decoded(&())
                     .map_err(crate::Error::from)
                     .map(Iterator::count)
-                    .map(|count| count == 1)
+                    .and_then(|count| {
+                        ensure!(count == 1, crate::Error::WalletWithoutARootKey);
```

Comment:
Please change the error name to `WalletSanityErrorInvalidRootKeyCount(usize, usize)`, and in the text specify expected vs current.

Think of this from the user's perspective. If this error (ever) happens, it's of interest to understand what the problem is to be able to track it. In the future, we may add more sanity checks and they should have different errors/error messages.

Reply:
The count should not bring much to the error message as it should only be 0 or 1.
I will replace the count with a get in this case to make it clearer, that we are just checking if the root keys exist or not, if not then we have a wallet file without root keys which should not happen.

Reply:
The current count at least helps because it can tell us whether there's a corruption in the wallet.

Reply:
The construction of the map itself restricts that, as the Key is an empty Struct which is unique and can't construct a map with more than 1 element.

Reply:
That's why I said "corruption". If the error says there are 2710223923 keys, then we know the db is broken. The assumption that everything is bound to our code is very optimistic. So in database-related things, we try to be a little bit pessimistic to account for corruption.